Index: tiopenmax-audiomanager-3.23/system/src/openmax_il/audio_manager/src/AudioManager.c
===================================================================
--- tiopenmax-audiomanager-3.23.orig/system/src/openmax_il/audio_manager/src/AudioManager.c	2009-08-19 18:17:26.000000000 +0200
+++ tiopenmax-audiomanager-3.23/system/src/openmax_il/audio_manager/src/AudioManager.c	2009-08-19 18:19:13.000000000 +0200
@@ -2840,6 +2840,8 @@
     system("/usr/bin/env amixer cset numid=1 1");
     /* Setting the input method so that encoders can work in dasf mode */
     system("/usr/bin/env amixer cset numid=31 1");
+    /* Setting apll power control to on */
+    system("/usr/bin/env amixer cset numid=20 1");
 #endif
 
 }
