Index: exmap-console/kernel/exmap.c
===================================================================
--- exmap-console.orig/kernel/exmap.c	2009-07-29 17:14:05.000000000 +0200
+++ exmap-console/kernel/exmap.c	2009-07-29 17:14:22.000000000 +0200
@@ -519,7 +519,6 @@
 	
 	exmap_proc_file->read_proc = procfile_read;
 	exmap_proc_file->write_proc = procfile_write;
-	exmap_proc_file->owner = THIS_MODULE;
 	
 	/*     exmap_proc_file->mode         = S_IFREG | S_IRUGO; */
 	/* TODO - this is quite probably a security problem */
