Index: L23-17_LO/meta/packages/netbase/netbase_4.21.bb
===================================================================
--- meta/packages/netbase/netbase_4.21.bb	2009-08-31 10:25:34.000000000 +0200
+++ meta/packages/netbase/netbase_4.21.bb	2009-08-31 10:25:45.000000000 +0200
@@ -38,7 +38,7 @@
 
 	# Disable network manager on machines that commonly do NFS booting
 	case "${MACHINE}" in
-		"omap-3430sdp" | "omap-3430ldp" | "omap-2430sdp" | "qemuarm" | "qemux86" )
+		"zoom2" | "zoom3" | "omap-3630sdp" | "omap-3430sdp" | "omap-3430ldp" | "omap-2430sdp" | "qemuarm" | "qemux86" )
 			touch ${D}${sysconfdir}/network/nm-disabled-eth0
 			;;
 		*)
