Index: mm-isp-1.15/linux/mm_isp/camera_algo_frmwk/src/Makefile
===================================================================
--- mm-isp-1.15.orig/linux/mm_isp/camera_algo_frmwk/src/Makefile	2009-07-17 22:43:26.000000000 +0200
+++ mm-isp-1.15/linux/mm_isp/camera_algo_frmwk/src/Makefile	2009-07-17 22:43:47.000000000 +0200
@@ -24,7 +24,7 @@
 
 include $(MMISPROOT)/Master.mk
 
-CFLAGS+=-I../inc -I$(MMISPINCLUDEDIR) -I$(KERNISPINCLUDEDIR)/include/asm-arm/arch-omap -I../../../../algo/camera/isphal/inc -Wall -fpic -pipe
+CFLAGS+=-I../inc -I$(MMISPINCLUDEDIR) -I$(KERNISPINCLUDEDIR)/arch/arm/plat-omap/include/mach -I../../../../algo/camera/isphal/inc -Wall -fpic -pipe
 ifeq ($(USE_HWP), 1)
 CFLAGS+=-DUSE_HWP=1
 else
Index: mm-isp-1.15/linux/mm_isp/il3p/src/Makefile
===================================================================
--- mm-isp-1.15.orig/linux/mm_isp/il3p/src/Makefile	2009-07-17 22:40:52.000000000 +0200
+++ mm-isp-1.15/linux/mm_isp/il3p/src/Makefile	2009-07-17 22:42:08.000000000 +0200
@@ -8,7 +8,7 @@
 
 FRMWRKINCLUDEDIR=$(MMISPROOT)/camera_algo_frmwk/inc
 
-CFLAGS+=-I../inc -I$(MMISPINCLUDEDIR) -I$(FRMWRKINCLUDEDIR) -I$(KERNISPINCLUDEDIR)/include/asm-arm/arch-omap -I../../../../algo/camera/isphal/inc
+CFLAGS+=-I../inc -I$(MMISPINCLUDEDIR) -I$(FRMWRKINCLUDEDIR) -I$(KERNISPINCLUDEDIR)/arch/arm/plat-omap/include/mach -I../../../../algo/camera/isphal/inc
 
 COMPONENT_NAME=ALG.IL3P
 COMPONENT_LIB=lib$(COMPONENT_NAME).so
