Index: mm-isp-1.15/linux/mm_isp/camera_algo_frmwk/inc/camdriver_interface.h
===================================================================
--- mm-isp-1.15.orig/linux/mm_isp/camera_algo_frmwk/inc/camdriver_interface.h	2009-04-27 23:54:43.000000000 +0200
+++ mm-isp-1.15/linux/mm_isp/camera_algo_frmwk/inc/camdriver_interface.h	2009-08-26 12:42:07.000000000 +0200
@@ -11,6 +11,8 @@
 
 #include <stdio.h>
 #include "stdtypes.h"
+#include <linux/types.h>
+#include <linux/videodev2.h>
 #include <isp_user.h>
 
 struct isptables_update {
