Index: tiopenmax-g729enc-3.23/audio/src/openmax_il/g729_enc/tests/Makefile
===================================================================
--- tiopenmax-g729enc-3.23.orig/audio/src/openmax_il/g729_enc/tests/Makefile	2009-07-07 12:01:32.000000000 +0200
+++ tiopenmax-g729enc-3.23/audio/src/openmax_il/g729_enc/tests/Makefile	2009-07-07 12:01:50.000000000 +0200
@@ -21,7 +21,7 @@
 
 include $(OMXROOT)/Master.mk
 
-CFLAGS+=-I../inc -I$(OMXINCLUDEDIR) -I$(SYSTEMINCLUDEDIR) -I$(BRIDGEINCLUDEDIR) -I$(GTTOOLSINCDIR) -Wall -ansi -fpic -pipe -DOMAP_2430 -DETEEDN_PATH
+CFLAGS+=-I../inc -I$(OMXINCLUDEDIR) -I$(SYSTEMINCLUDEDIR) -I$(BRIDGEINCLUDEDIR) -I$(GTTOOLSINCDIR) -Wall -fpic -pipe -DOMAP_2430 -DETEEDN_PATH
 
 ifeq ($(OMX_DEBUG), 1)
     CFLAGS += -O0
