Index: git/drivers/input/touchscreen/ads7846.c
===================================================================
--- git.orig/drivers/input/touchscreen/ads7846.c	2009-08-11 17:45:34.000000000 +0200
+++ git/drivers/input/touchscreen/ads7846.c	2009-08-11 17:47:53.000000000 +0200
@@ -28,7 +28,7 @@
 #include <linux/spi/spi.h>
 #include <linux/spi/ads7846.h>
 #include <asm/irq.h>
-
+#include <linux/i2c/twl4030.h>
 
 /*
  * This code has been heavily tested on a Nokia 770, and lightly
@@ -1242,8 +1242,16 @@
 	.resume		= ads7846_resume,
 };
 
+/* Enabling VAUX3 LDO used by Touchscreen */
+static void enable_vaux3(void)
+{
+        twl4030_i2c_write_u8(TWL4030_MODULE_PM_RECEIVER,
+                        0x20, 0x1F);
+}
+
 static int __init ads7846_init(void)
 {
+	enable_vaux3();
 	return spi_register_driver(&ads7846_driver);
 }
 module_init(ads7846_init);
