Index: gst-plugins-base-0.10.20/gst/audioconvert/gstchannelmix.c
===================================================================
--- gst-plugins-base-0.10.20.orig/gst/audioconvert/gstchannelmix.c	2009-05-06 11:30:37.000000000 -0500
+++ gst-plugins-base-0.10.20/gst/audioconvert/gstchannelmix.c	2009-05-06 11:30:59.000000000 -0500
@@ -636,7 +636,7 @@
     g_string_append (s, " }");
   }
   g_string_append (s, " }");
-  GST_DEBUG (s->str);
+  GST_DEBUG ("%s", s->str);
   g_string_free (s, TRUE);
 }
 
