Index: tiopenmax-videoenc-3.23-pre/video/src/openmax_il/video_encode/test/Makefile
===================================================================
--- tiopenmax-videoenc-3.23-pre.orig/video/src/openmax_il/video_encode/test/Makefile	2009-06-15 23:45:43.000000000 -0500
+++ tiopenmax-videoenc-3.23-pre/video/src/openmax_il/video_encode/test/Makefile	2009-06-17 15:21:44.000000000 -0500
@@ -17,14 +17,13 @@
     CFLAGS += -DVIDENCTEST_DEBUG 
 endif
 
-OMXCORE_LIB=$(OMXLIBDIR)/libOMX_Core.so
 COMPONENT_TEST=VideoEncTest
 
 COMPONENT_TARGET=$(OMXTESTDIR)/$(COMPONENT_TEST)
 
 all install:: $(COMPONENT_TARGET)
 
-omxcore $(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR) $(OMXCORE_LIB)  
+$(COMPONENT_TARGET): $(OMXTESTDIR) $(COMPONENT_TEST) $(OMXLIBDIR)
 	@echo "Installing $(COMPONENT_TEST)"
 	cp -f $(COMPONENT_TEST) $(COMPONENT_TARGET)
 
