Index: tidspbridge-module-23.17/mpu_driver/make/start.mk
===================================================================
--- tidspbridge-module-23.17.orig/mpu_driver/make/start.mk	2009-06-18 16:35:04.000000000 +0200
+++ tidspbridge-module-23.17/mpu_driver/make/start.mk	2009-06-18 16:35:19.000000000 +0200
@@ -31,7 +31,7 @@
 $(error Error: variable PROJROOT not defined)
 endif
 
-CMDDEFS =
+CMDDEFS ?=
 CMDDEFS_START =
 
 # this is a driver - so find the kernel 
