Index: tiopenmax-ram-3.23/system/src/openmax_il/resource_manager/resource_activity_monitor/src/Resource_Activity_Monitor.c
===================================================================
--- tiopenmax-ram-3.23.orig/system/src/openmax_il/resource_manager/resource_activity_monitor/src/Resource_Activity_Monitor.c	2009-08-05 18:22:40.000000000 +0200
+++ tiopenmax-ram-3.23/system/src/openmax_il/resource_manager/resource_activity_monitor/src/Resource_Activity_Monitor.c	2009-08-05 18:22:49.000000000 +0200
@@ -50,7 +50,7 @@
     printf("[setting operating point] MHz = %d operatingPoint = %d\n",MHz,operatingPoint);
     strcpy(command,"echo -n ");
     strcat(command,ram_itoa(operatingPoint));
-    strcat(command," > /sys/power/vdd1_opp_value");
+    strcat(command," > /sys/power/vdd1_opp");
     system(command);
 #endif    
 }
