Index: tiopenmax-imaadpcmenc-3.22/audio/src/openmax_il/imaadpcm_enc/tests/Makefile
===================================================================
--- tiopenmax-imaadpcmenc-3.22.orig/audio/src/openmax_il/imaadpcm_enc/tests/Makefile
+++ tiopenmax-imaadpcmenc-3.22/audio/src/openmax_il/imaadpcm_enc/tests/Makefile
@@ -23,7 +23,7 @@ OBJ=$(SRC:.c=.o)
 
 include $(OMXROOT)/Master.mk
 
-CFLAGS+=-I../inc -I$(OMXINCLUDEDIR) -I$(SYSTEMINCLUDEDIR) -I$(BRIDGEINCLUDEDIR) -I$(GTTOOLSINCDIR) -Wall -ansi -fpic -pipe -DOMAP_2430 -DETEEDN_PATH -DDSP_RENDERING_ON
+CFLAGS+=-I../inc -I$(OMXINCLUDEDIR) -I$(SYSTEMINCLUDEDIR) -I$(BRIDGEINCLUDEDIR) -I$(GTTOOLSINCDIR) -Wall -fpic -pipe -DOMAP_2430 -DETEEDN_PATH -DDSP_RENDERING_ON
 
 ifeq ($(OMX_DEBUG), 1)
     CFLAGS += -O0
